lib/ash_state_machine.ex

```diff
@@ -111,6 +111,7 @@ defmodule AshStateMachine do
   use Spark.Dsl.Extension,
     sections: @sections,
     transformers: [
+      AshStateMachine.Transformers.SetDefaultInitialState,
```

Comment:
This should be run before the AddState transformer and putting it above it seems to do that, but I tried to add an after?/before? but that lead to spark giving me a transformer cycle error. not sure why though. AddState has before and after as well, not sure why that makes a difference though.

```elixir
  def before?(Ash.Resource.Transformers.AddState), do: true
  def before?(_), do: false

  def after?(Ash.Resource.Transformers.AddState), do: false
  def after?(_), do: true
```